backend/plugin/parser/cassandra/query_span_extractor.go

```diff
@@ -217,15 +228,137 @@ func (e *querySpanExtractor) expandSelectAsterisk(keyspace, table string) []base
+			Table:    table,
+			Column:   columnName,
+		}
+
+		// Add to both SourceColumns and PredicateColumns
+		e.querySpan.SourceColumns[colResource] = true
+		e.querySpan.PredicateColumns[colResource] = true
+	}
+
+	// TODO: Handle more complex cases:
+	// - Functions containing columns (e.g., token(column))
+	// - Collection operations (e.g., collection CONTAINS value)
+	// - Nested expressions
+}
+
+// EnterInsert is called when we enter an INSERT statement
+func (e *querySpanExtractor) EnterInsert(_ *cql.InsertContext) {
+	if e.err != nil {
+		return
+	}
+	// Set query type to DML for INSERT
+	e.querySpan.Type = base.DML
+}
+
+// EnterUpdate is called when we enter an UPDATE statement
+func (e *querySpanExtractor) EnterUpdate(_ *cql.UpdateContext) {
+	if e.err != nil {
+		return
+	}
+	// Set query type to DML for UPDATE
+	e.querySpan.Type = base.DML
+}
+
+// EnterDelete_ is called when we enter a DELETE statement
+func (e *querySpanExtractor) EnterDelete_(_ *cql.Delete_Context) {
+	if e.err != nil {
+		return
+	}
+	// Set query type to DML for DELETE
+	e.querySpan.Type = base.DML
+}
 
-	// This is a simplified implementation - a complete one would
-	// walk through all relation elements and extract column names
+// DDL Statement Handlers
+
+// EnterCreateTable is called when we enter a CREATE TABLE statement
+func (e *querySpanExtractor) EnterCreateTable(_ *cql.CreateTableContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterAlterTable is called when we enter an ALTER TABLE statement
+func (e *querySpanExtractor) EnterAlterTable(_ *cql.AlterTableContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterDropTable is called when we enter a DROP TABLE statement
+func (e *querySpanExtractor) EnterDropTable(_ *cql.DropTableContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterCreateKeyspace is called when we enter a CREATE KEYSPACE statement
+func (e *querySpanExtractor) EnterCreateKeyspace(_ *cql.CreateKeyspaceContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterAlterKeyspace is called when we enter an ALTER KEYSPACE statement
+func (e *querySpanExtractor) EnterAlterKeyspace(_ *cql.AlterKeyspaceContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterDropKeyspace is called when we enter a DROP KEYSPACE statement
+func (e *querySpanExtractor) EnterDropKeyspace(_ *cql.DropKeyspaceContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterCreateIndex is called when we enter a CREATE INDEX statement
+func (e *querySpanExtractor) EnterCreateIndex(_ *cql.CreateIndexContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
+}
+
+// EnterDropIndex is called when we enter a DROP INDEX statement
+func (e *querySpanExtractor) EnterDropIndex(_ *cql.DropIndexContext) {
+	if e.err != nil {
+		return
+	}
+	e.querySpan.Type = base.DDL
 }
```

Comment:
alterMaterializedView, alterType, createFunction, createTrigger, createMaterializedView and so on is DDL I thought even though the [doc](https://cassandra.apache.org/doc/4.0/cassandra/cql/ddl.html) doesn't mention them.